### ISSUE TITLE
Replace alias datatypes by their standard variants

### ISSUE THREAD

**Issue by jougs** [nest/nest-simulator#343]
We're often using our own typedef'd versions of standard datatypes throughout the code (e.g. `int_t` instead of just `int`; see line 104-109 in [nest_types.h](https://github.com/nest/nest-simulator/blob/dc5edb5808c2526229acbb3989a54acb4efd8d29/nestkernel/nest_types.h#L104))

The use is inconsistently and brings more confusion than readability. These custom types should be removed and their usage replaced by the standard types.

This issue was known as trac90. See also #344.

**Comment by sanjayankur31** [comment]
@jougs - has a decision been made on this? Does someone just need to sit down and remove the aliases in favour of their standard variants now?

**Comment by sanjayankur31** [comment]
The advantages I see of using aliases are:

- readability (`weight` makes more sense than `double_t`)
- aliases are usually shorter (`somealias` instead of `unsigned long long` and so on)

**Comment by heplesser** [comment]
@sanjayankur31 This issue is only about the following aliases for standard C++ datatypes:
```
typedef double double_t;       ///< Double precision floating point numbers.
typedef float float_t;         ///< Single precision floating point numbers.
typedef int int_t;             ///< Integer number with at least 16 bit.
typedef long long_t;           ///< Integer number with at least 32 bit.
typedef unsigned int uint_t;   ///< Unsigned int_t.
typedef unsigned long ulong_t; ///< Unsigned long_t.
```
At some point very long ago we had an idea that you could change `double_t` to be an alias for `float` and thus reduce the memory required for network components. But that would have required fully consistent use of `double_t` throughout, which we do not have at all. We also learned since that memory issues are elsewhere (see eg Kunkel et al, 2014). Therefore, these aliases are mostly confusing.

I will put this on the agenda for next Monday's developer meeting, but unless there should be serious objections raised there, I think we should implement this soon.

**Comment by heplesser** [comment]
In the  [Open NEST Developer VC today](https://github.com/nest/nest-simulator/wiki/2016-08-08-Open-NEST-Developer-Video-Conference) we decided to remove all the `_t` datatypes.

**Comment by heplesser** [comment]
Closing as this has been fixed by#454.
